reflect/src/main/java/dagger/reflect/ComponentInvocationHandler.java

```diff
@@ -86,9 +89,18 @@ private ComponentInvocationHandler(Scope scope) {
     }
 
     if (args == null || args.length == 0) {
-      if (returnType instanceof Class<?>
-          && ((Class<?>) returnType).getAnnotation(Subcomponent.class) != null) {
-        throw notImplemented("Subcomponents");
+      if (returnType instanceof Class<?>) {
+        Class<?> returnClass = (Class<?>) returnType;
+        if (returnClass.getAnnotation(Subcomponent.class) != null) {
+          return new SubcomponentMethodInvocationHandler(returnClass, scope);
+        }
+        // TODO if (returnClass.getAnnotation(Subcomponent.Builder.class) != null) {
+        // Until @Subcomponent.Builder has runtime retention, check whether this return type belongs
+        // to a subcomponent builder by looking for an enclosing class annotated with @Subcomponent.
```

Comment:
Already fixed internally at Google. Waiting for sync out to GitHub. Also fixed `@Component.Builder` and some other annotations so we can use them.